server/channels/api4/channel.go

```diff
@@ -377,9 +377,17 @@ func patchChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	if patch.BannerInfo != nil && (originalOldChannel.Type != model.ChannelTypeOpen && originalOldChannel.Type != model.ChannelTypePrivate) {
-		c.Err = model.NewAppError("patchChannel", "api.channel.update_channel.banner_info.channel_type.not_allowed", nil, "", http.StatusBadRequest)
-		return
+	if patch.BannerInfo != nil {
```

Comment:
nit: extract this logic to a validation to a helper function. It could make the code clearer about the intention of this logic.

Reply:
Done.